lib/test_prof/tag_prof/minitest.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module Minitest
+  module TestProf
+    class TagProfReporter < BaseReporter # :nodoc:
+      attr_reader :results
+
+      def initialize(io = $stdout, _options = {})
+        super
+        @results = ::TestProf::TagProf::Result.new("type")
+
+        if event_prof_activated?
+          require "test_prof/event_prof"
+          @current_group_id = nil
+          @events_profiler = configure_profiler
+          @results = ::TestProf::TagProf::Result.new("type", @events_profiler.events)
+        end
+      end
+
+      def prerecord(group, example)
+        return unless event_prof_activated?
+
+        # enable event profiling
+        @events_profiler.group_started(true)
+      end
+
+      def record(result)
+        results.track(main_folder_path(result), time: result.time, events: fetch_events_data)
+        @events_profiler.group_started(nil) if event_prof_activated? # reset and disable event profilers
+      end
+
+      def report
+        printer = (ENV["TAG_PROF_FORMAT"] == "html") ? ::TestProf::TagProf::Printers::HTML : ::TestProf::TagProf::Printers::Simple
+        printer.dump(results)
+      end
+
+      private
+
+      def main_folder_path(result)
+        return "subdirectory_not_found: #{result.source_location.first}" if absolute_path_from(result).nil?
```

Comment:
I'm not sure if this is the right behavior. Maybe we want to raise if the root test directory cannot be found ?

Reply:
Answered above: let's return `:__unknown__`